test/test_pyega3.py

```diff
@@ -630,6 +650,246 @@ def test_load_server_config_missingAttributesInJsonFile(self):
+                                        file_size_with_iv, 'check_sum', 1, None, output_file_path, None, 2, 0.1)
+
+        temp_file = pyega3.TEMPORARY_FILES.pop()
+        # The temporary file should not exist because everything went fine,
+        # and it was deleted automatically:
+        self.assertFalse(os.path.exists(temp_file))
+
+        self.assertTrue(os.path.exists(output_file_path))
+        output_file_size = os.stat(output_file_path).st_size
+        self.assertEqual(output_file_size, file_size_without_iv)
+        os.remove(output_file_path)
+
+    @responses.activate
+    def test_second_attempt_succeeds(self):
+        """
+        It was not possible to download the whole file on the first download attempt,
+        so the script retries for a second time and continues from where it stopped
+        on the first attempt.
+        """
+
+        pyega3.TEMPORARY_FILES_SHOULD_BE_DELETED = False
+
+        file_size_without_iv = 92700
+        file_size_with_iv = file_size_without_iv + 16
+
+        self.server_config_file_loaded_successfully()
+        self.user_has_authenticated_successfully()
+
+        amount_of_missing_bytes = 123
+        file_size_with_missing_bytes = file_size_without_iv - amount_of_missing_bytes
+        input_file_with_few_bytes_missing = bytearray(os.urandom(file_size_with_missing_bytes))
+        self.file_can_be_downloaded(input_file_with_few_bytes_missing)
+
+        rest_of_the_input_file = bytearray(os.urandom(amount_of_missing_bytes))
+        self.file_can_be_downloaded(rest_of_the_input_file)
+
+        output_file_path = self.create_output_file_path()
+
+        pyega3.download_file_retry(('', ''), 'test_file_id1', output_file_path, output_file_path,
+                                   file_size_with_iv, 'check_sum', 1, None, output_file_path, None, 2, 0.1)
+
+        temp_file = pyega3.TEMPORARY_FILES.pop()
+        # The temporary file should not exist because everything went fine,
+        # and it was deleted automatically:
+        self.assertFalse(os.path.exists(temp_file))
+
+        self.assertEqual(responses.calls[1].request.headers.get('Range'), 'bytes=0-92699')
+        self.assertEqual(responses.calls[2].request.headers.get('Range'), 'bytes=92577-92699')
+        self.assertEqual(responses.calls[2].request.headers.get('Range'), 'bytes={}-92699'
+                         .format(file_size_with_missing_bytes))
+
+        self.assertTrue(os.path.exists(output_file_path))
+        output_file_size = os.stat(output_file_path).st_size
+        self.assertEqual(output_file_size, file_size_without_iv)
+        os.remove(output_file_path)
+
+    def test_deleting_non_existent_file_does_not_raise_exception(self):
+        non_existent_file = '/tmp/non/existent/file'
+        self.assertFalse(os.path.exists(non_existent_file))
+
+        # No exception is raised:
+        pyega3.delete_temporary_files([non_existent_file])
```

Comment:
In the code, it looks like the pyega3 raises an exception(FileNotFoundError) in delete_temporary_files, so why it is not generating the error here?

Reply:
Yes, you're right!

[On line 533 in the pyega3.py file](https://github.com/EGA-archive/ega-download-client/pull/91/files#diff-0621ff1c3cfeec9a76491eb78a70e253R533), the FileNotFoundError exception is caught (and then logged), so the FileNotFoundError exception never leaves the delete_temporary_files method.

This test tests exactly that fact: that the pyega3.delete_temporary_files raises no FileNotFoundError exception. Therefore the test is called test_deleting_non_existent_file_does_not_raise_exception ([on line 770](https://github.com/EGA-archive/ega-download-client/pull/91/files#diff-ad62edcee79d3113ca926e0b183b735aR770)) and the test has a comment ([on line 774](https://github.com/EGA-archive/ega-download-client/pull/91/files#diff-ad62edcee79d3113ca926e0b183b735aR774)) that "No exception is raised".

Does this answer your question or did I misunderstand something?